csvqb/csvqb/tests/unit/cube/qb/errorvalidation.py

```diff
@@ -61,6 +62,28 @@ def test_multi_measure_qb_definition(self):
 
         self.assert_no_validation_errors(validation_errors)
 
+    def test_existing_dimension_output_uri_template(self):
+        """
+        An ExistingQbDimension must have an output_uri_template defined by the user if not it's an error
+
+        """
+
+        data = pd.DataFrame({
+            "Existing Dimension": ["A", "B", "C"],
+            "Value": [1, 2, 3]
+        })
+        cube = Cube(CubeMetadata("Cube's name"), data, [
+            QbColumn("Existing Dimension", ExistingQbDimension("http://example.org/dimensions/location")),
+            QbColumn("Value",
+                     QbSingleMeasureObservationValue(ExistingQbUnit("http://some/unit"),
+                                                     ExistingQbMeasure("http://some/measure")))
+        ])
+
+        errors = cube.validate()
+        errors += validate_qb_component_constraints(cube)
+
+        self.assertEqual(1, len(errors))
```

Comment:
I think we should be more specific about the error we expect - i.e. check that the text matches to some degree. Otherwise it could be another error we're triggering and we'd never know.

Reply:
Expected error message is asserted and the error message matches to what we are trying to test.

Reply:
Looks good to me. Thanks!